core/src/main/java/com/amplifyframework/auth/AuthException.java

```diff
@@ -191,4 +191,384 @@ public UnknownException(Throwable cause) {
+            public InvalidPasswordException(Throwable cause) {
+                super(MESSAGE, cause, RECOVERY_SUGGESTION);
+            }
+        }
+
+        /**
+         * Could not perform the action because number of allowed operation has exceeded.
+         */
+        public class LimitExceededException extends AWSCognitoAuthException {
+            private static final long serialVersionUID = 1L;
+            private static final String MESSAGE = "Number of allowed operation has exceeded.";
+            private static final String RECOVERY_SUGGESTION = "Turn off and re-attempt this operation.";
+
+            /**
+             * Default message/recovery suggestion with a cause.
+             * @param cause The original error.
+             */
+            public LimitExceededException(Throwable cause) {
+                super(MESSAGE, cause, RECOVERY_SUGGESTION);
+            }
+        }
+
+        /**
+         * Could not perform the action because password needs to be reset.
+         */
+        public static class PasswordResetRequiredException extends AWSCognitoAuthException {
+            private static final long serialVersionUID = 1L;
+            private static final String MESSAGE = "Required to reset the password of the user.";
+            private static final String RECOVERY_SUGGESTION = "Reset the password of the user.";
+
+            /**
+             * Default message/recovery suggestion with a cause.
+             * @param cause The original error.
+             */
+            public PasswordResetRequiredException(Throwable cause) {
+                super(MESSAGE, cause, RECOVERY_SUGGESTION);
+            }
+        }
+
+        /**
+         * Could not perform the action because Amazon Cognito service cannot find the requested resource.
+         */
+        public static class ResourceNotFoundException extends AWSCognitoAuthException {
+            private static final long serialVersionUID = 1L;
+            private static final String MESSAGE = "Amazon Cognito service cannot find the requested resource.";
+            private static final String RECOVERY_SUGGESTION = "Turn off and re-attempt this operation.";
+
+            /**
+             * Default message/recovery suggestion with a cause.
+             * @param cause The original error.
+             */
+            public ResourceNotFoundException(Throwable cause) {
+                super(MESSAGE, cause, RECOVERY_SUGGESTION);
+            }
+        }
+
+        /**
+         * Could not perform the action because user made too many failed attempts for a given action.
+         */
+        public static class FailedAttemptsLimitExceededException extends AWSCognitoAuthException {
+            private static final long serialVersionUID = 1L;
+            private static final String MESSAGE = "User has made too many failed attempts for a given action.";
+            private static final String RECOVERY_SUGGESTION = "Turn off and re-attempt this operation.";
```

Comment:
This recovery suggestion is wrong - for this type of error we may want to let the developer specify the recovery suggestion since it depends on the way the service is configured. Is the account now locked until a password reset? Is there a cool off period? If so what is the cool off period?

Reply:
So in this case I'll change the recovery suggestion to "Please check out the service configuration to see the condition of locking."